File: CSharpCodeBase/utils/hump/timer.cs
using System;
 
 using System.Collections.Generic;
 
 using System.Linq;
 
 using System.Text;
 
 using UnityEngine;
 
 using System.Collections;
 
 using MainGame.core;
 
 using UnityEngine.EventSystems;
 
 namespace MainGame{
 public class timer {
 public void init(self, timeout);
       this.timeout = timeout;
       this._original_timeout = timeout;
 }})
 public void Tick(){
   this.timeout = this.timeout - GameController.deltaTime;
   return this.timeout < 0;
 }
 public void Reset(){
   this.timeout = this._original_timeout;
 }
 }}

File: CSharpCodeBase/utils/hump/timer.lua
local Class = require 'utils.hump.class'

local Timer = Class({init=function(self, timeout)
      self.timeout = timeout
      self._original_timeout = timeout
end})

function Timer:Tick()
  self.timeout = self.timeout - GameController.deltaTime
  return self.timeout < 0
end

function Timer:Reset()
  self.timeout = self._original_timeout
end

return Timer